src/bin/nydusd/fs_cache.rs

```diff
@@ -416,16 +418,52 @@ impl FsCacheHandler {
             match self.create_data_blob_object(&config, msg.fd) {
                 Err(s) => format!("copen {},{}", hdr.msg_id, s),
                 Ok((blob, blob_size)) => {
-                    state
-                        .id_to_object_map
-                        .insert(hdr.object_id, (FsCacheObject::DataBlob(blob), msg.fd));
+                    state.id_to_object_map.insert(
+                        hdr.object_id,
+                        (FsCacheObject::DataBlob(blob.clone()), msg.fd),
+                    );
                     state.id_to_config_map.insert(hdr.object_id, config.clone());
+                    let _ = self.do_prefetch(&config, blob.clone());
                     format!("copen {},{}", hdr.msg_id, blob_size)
                 }
             }
         }
     }
 
+    pub fn do_prefetch(
+        //prefetch_files: Option<Vec<PathBuf>>,
+        &self,
+        config: &BlobCacheConfigDataBlob,
+        blob: Arc<dyn BlobCache>,
+    ) {
+        let blob_info = config.blob_info().deref().clone();
+        let factory_config = config.factory_config().deref().clone();
+        if factory_config.cache.prefetch_config.enable {
+            let size = factory_config.cache.prefetch_config.merging_size as u64;
+            let count = blob_info.compressed_size() / size;
+            let mut blob_rep = Vec::with_capacity(count as usize);
+            let mut pre_offset = blob_info.readahead_offset();
```

Comment:
Seems you are trying to prefetch all data of the blob, so why using `blob_info.readahead_offset()` instead of `0` here?